pkg/controller/clusteringress/clusteringress_controller.go

```diff
@@ -103,6 +103,12 @@ func (r *ReconcileClusterIngress) Reconcile(request reconcile.Request) (reconcil
 
 	// Don't modify the informer's copy
 	ci := original.DeepCopy()
+	if newFinalizer, change := r.base.AppendIfAbsent(ci.Finalizers, "ocp-ingress"); change {
```

Comment:
Should we add an ‚ensureFinalizer‘ function to the bse reconciler like in the reference?

Reply:
If we add ensureFinalizer we need to use `ensureFinalizer` wherever `AppendIfAbsent` exist
because logic is same